lib/class-wp-rest-menu-items-controller.php

```diff
@@ -0,0 +1,658 @@
+		 *
+		 * @param object          $nav_menu_item Inserted or updated nav item object.
+		 * @param WP_REST_Request $request       Request object.
+		 * @param bool            $creating      True when creating a post, false when updating.
+		 */
+		do_action( "rest_after_insert_{$this->post_type}", $nav_menu_item_id, $request, true );
+
+		$response = $this->prepare_item_for_response( $nav_menu_item_id, $request );
+		$response = rest_ensure_response( $response );
+
+		$response->set_status( 201 );
+		$response->header( 'Location', rest_url( sprintf( '%s/%s/%d', $this->namespace, $this->rest_base, $nav_menu_item_id ) ) );
+
+		return $response;
+	}
+
+	/**
+	 * Updates a single nav menu item.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$valid_check = $this->get_post( $request['id'] );
+		if ( is_wp_error( $valid_check ) ) {
+			return $valid_check;
+		}
+
+		$prepared_nav_item = $this->prepare_item_for_database( $request );
+
+		if ( is_wp_error( $prepared_nav_item ) ) {
+			return $prepared_nav_item;
+		}
+
+		$menu_id = (int) $request['menu_id'];
+
+		$nav_menu_item_id = wp_update_nav_menu_item( $menu_id, $request['id'], $prepared_nav_item );
+
+		if ( is_wp_error( $nav_menu_item_id ) ) {
+			if ( 'db_update_error' === $nav_menu_item_id->get_error_code() ) {
+				$nav_menu_item_id->add_data( array( 'status' => 500 ) );
+			} else {
+				$nav_menu_item_id->add_data( array( 'status' => 400 ) );
+			}
+
+			return $nav_menu_item_id;
+		}
+
+		$nav_menu_item = $this->get_nav_menu_item( $nav_menu_item_id );
+
+		/** This action is documented in wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php */
+		do_action( "rest_insert_{$this->post_type}", $nav_menu_item, $request, false );
+
+		$schema = $this->get_item_schema();
+
+		if ( ! empty( $schema['properties']['meta'] ) && isset( $request['meta'] ) ) {
+			$meta_update = $this->meta->update_value( $request['meta'], $post->ID );
+
+			if ( is_wp_error( $meta_update ) ) {
+				return $meta_update;
+			}
+		}
+
+		$nav_menu_item = $this->get_nav_menu_item( $nav_menu_item_id );
+		$fields_update = $this->update_additional_fields_for_object( $nav_menu_item, $request );
+
+		if ( is_wp_error( $fields_update ) ) {
+			return $fields_update;
+		}
+
+		$request->set_param( 'context', 'edit' );
+
+		/** This action is documented in wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php */
+		do_action( "rest_after_insert_{$this->post_type}", $nav_menu_item, $request, false );
+
+		$response = $this->prepare_item_for_response( $nav_menu_item, $request );
+
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Prepares a single post for create or update.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return stdClass|WP_Error Post object or WP_Error.
```

Comment:
This is now returning an `array`. Would it be better to return an `stdClass` and cast to an array when calling `wp_update_nav_menu_item()`. That would be consistent with the rest of our controllers.